atc/path_flag.go

```diff
@@ -2,19 +2,39 @@ package atc
 
 import (
 	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
 	"path/filepath"
 	"strings"
 
 	"github.com/jessevdk/go-flags"
 )
 
+const tempFilePattern = "___dash-stdin-path-flag___"
+
 type PathFlag string
 
 func (path *PathFlag) UnmarshalFlag(value string) error {
 	if value == "" {
 		return nil
 	}
 
+	if value == "-" {
+		tempf, err := ioutil.TempFile("", tempFilePattern)
+		if err != nil {
+			return fmt.Errorf("failed to create a temp file")
+		}
+		defer tempf.Close()
+
+		_, err = io.Copy(tempf, os.Stdin)
+		if err != nil {
+			return fmt.Errorf("failed to write temp file: %s", err.Error())
+		}
+		*path = PathFlag(tempf.Name())
+		return nil
+	}
```

Comment:
It might be possible on Linux and Mac to get away with
```go
if value == "-" {
  *path = PathFlag("/dev/stdin")
  return nil
}
```

At least, I can do `fly -t old gp -p foo | fly -t new sp -p foo -n -c /dev/stdin` with current `fly`, and it works; I didn't test making that change in `PathFlag`. I'm pretty sure this will not work on Windows though. Do you think it's worth having this simpler way if `GOOS` is non-Windows?

Reply:
Good point. TBH, I always ignore Windows because I don't have a windows env in hand. I'll add a non-windows check.